pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -417,25 +417,39 @@ func (m *kubeGenericRuntimeManager) podSandboxChanged(pod *v1.Pod, podStatus *ku
 		}
 	}
 
-	// Needs to create a new sandbox when readySandboxCount > 1 or the ready sandbox is not the latest one.
+	//check the pod is succeeded
+	succeededContainerCount := 0
+	for _, container := range pod.Spec.Containers {
+		for _, status := range podStatus.ContainerStatuses {
+			if container.Name == status.Name {
+				if status.State == kubecontainer.ContainerStateExited && status.ExitCode == 0 {
+					succeededContainerCount++
+				}
+			}
+		}
+	}
```

Comment:
Move this block after the readySandboxCount check

Reply:
Ok,I have move this block after the readySandboxCount check.